src/queryExecutionContext/pipelinedQueryExecutionContext.ts

```diff
@@ -72,6 +74,15 @@ export class PipelinedQueryExecutionContext implements ExecutionContext {
     if (typeof limit === "number" && typeof offset === "number") {
       this.endpoint = new OffsetLimitEndpointComponent(this.endpoint, offset, limit);
     }
+
+    // If distinct then add that to the pipeline
+    const distinctType = partitionedQueryExecutionInfo.queryInfo.distinctType;
```

Comment:
Can we move this to the factory method of a distinct document query execution component base class. Ordered vs unordered is an implementation detail for `DISTINCT` and the pipeline should not really concern itself with it. If the distinct component wanted to it can new up an unordered componenet for every query if it wanted to.

Reply:
Prefer to keep this explicit for now. Trying to come back and refactor the query pipeline later and would like to have less inheritance and complexity